src/components/Buttons/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,53 @@
+import { SetStateAction } from "react";
+import style from "./Checkbox.module.scss";
+
+interface ICheckbox {
+  label: string;
+  title: string;
+  setRegisterValue: React.Dispatch<SetStateAction<string | boolean | string[] | null>>;
+  registerValue: string | boolean | string[] | null;
+}
+
+const Checkbox = ({ label, title, setRegisterValue, registerValue }: ICheckbox) => {
+  
+  const handleOnCheck = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const isChecked = e.target.checked;
+    if (title === "Spreads") {
+      if (Array.isArray(registerValue) && isChecked) {
+        setRegisterValue([...registerValue, label]);
+      } 
+      else if (Array.isArray(registerValue) && !isChecked) {
+        const filteredRegisterValue = registerValue.filter(value => value !== label)
+        setRegisterValue([...filteredRegisterValue])
+      }
+      else {
+        setRegisterValue([label]);
+      }
+    } 
+    else if (title === "Topping" && isChecked) {
+      setRegisterValue(label);
```

Comment:
**issue:** This `ifology` will be very hard to maintain. Also, there are titles hardcoded which makes this component not possible to scale. Let's refactor this so you don't have these ifs